### PR TITLE
Add Owncast

### PR DESCRIPTION
**Pull request by PsychoZander**
Add an egg for https://owncast.online. Could do with an addition to the install script to scrape and find latest version instead of set version.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### New Server Submissions:

1. [X] Does your submission pass tests (server is connectable)?
2. [ ] Does your server use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [X] Have you added the server to the main README.md?
4. [X] Have you added a unique README.md for the server you are adding?

### CODE REVIEW

File: software/owncast/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+If you want to reset the server completly, remove the `data` directory manually before reinstalling.
+
+### Server Ports
+
+Ports required to run the server in a table format.
+
+| Port            | default |
+| --------------- | ------- |
+| Webserver 	  | 8080    |
```
Comment (QuintenQVD0) [inline software/owncast/README.md:50]: By default the 8080 is the wings port. So for most people that port will already be in use

Comment (gOOvER) [reply software/owncast/README.md:50]: yes, i would change this; else a lot of people ask for support :)

Comment (PsychoZander) [reply software/owncast/README.md]: Updated in the docs to 8090 now.

File: software/owncast/egg-owncast-online.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            "container": "debian:buster-slim",
+            "entrypoint": "bash"
+        }
+    },
+    "variables": [
+        {
+            "name": "RTMP Port",
+            "description": "The port that is used to receive the stream data.",
+            "env_variable": "RTMP_PORT",
+            "default_value": "1935",
```
[the remark this answers is not in the document]
Comment (PsychoZander) [reply software/owncast/egg-owncast-online.json]: Update to 8091 now.

[the remark this answers is not in the document]
Comment (Software-Noob) [reply software/owncast/egg-owncast-online.json]: > I think, this Port will not work in Ptero. I mean you must choose a higher Port

Any port above 1024 will work as it doesn't require escalated privileges

[the remark this answers is not in the document]
Comment (gOOvER) [reply software/owncast/egg-owncast-online.json]: > > I think, this Port will not work in Ptero. I mean you must choose a higher Port
> 
> Any port above 1024 will work as it doesn't require escalated privileges

I wasn't sure about the port. Thanks for the clarification

**Comment by Software-Noob** [comment]
> Could do with an addition to the install script to scrape and find latest version instead of set version.

We have [scripts ](https://github.com/parkervcp/eggs/tree/master/scripts)to ease the creation of eggs, including a Github release grabber.

**Comment by PsychoZander** [comment]
> 

Including this in the install script now. For the Match variable, what would I use for the default? The filename is in the format: 

owncast-0.0.11-linux-64bit.zip

So I think I want to ensure I pull the 64bit linux version, so I would set the match to "-linux-64bit.zip"?

**Comment by gOOvER** [comment]
> > 
> 
> Including this in the install script now. For the Match variable, what would I use for the default? The filename is in the format:
> 
> owncast-0.0.11-linux-64bit.zip
> 
> So I think I want to ensure I pull the 64bit linux version, so I would set the match to "-linux-64bit.zip"?

`-linux-64bit `should be enough

File: software/owncast/egg-owncast-online.json [hunk 3]
```diff
@@ -0,0 +1,95 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v1",
+        "update_url": null
+    },
+    "exported_at": "2022-03-31T11:32:42+01:00",
+    "name": "owncast.online",
```
Comment (parkervcp) [inline software/owncast/egg-owncast-online.json:8]: This can probably just be Owncast or something similar.

**Comment by parkervcp** [comment]
Other than my single comment which is not a request for change this is acceptable. If someone can verify that it works that would be great.

**Review by gOOvER — APPROVED** [review]
- server was installed and running
- admin login works
- streaming via OBS works

Egg works fine

**Comment by PsychoZander** [comment]
I can also confirm that the update via re-install is functional. There has been a new update since I created the egg, and running the reinstall script correctly pulls, downloads and installs the latest version from the project GitHub repo.

**Review by parkervcp — APPROVED** [review]

**Pull request merged**